doc/development/releasing.md

```diff
@@ -0,0 +1,119 @@
+# Releasing
```

Comment:
do externals publish this repo?
If not, why not link the confluence page?

If you want to have it public. How about adding it to the release robot as "manual release documentation" and link it from different repositories.

Reply:
No, not really. To the sonatype group only people `@exasol.com` can publish. But the setup requires certain preparation. I would still keep it public for now.

I have not used RR so far. I already have TODO for it, then this doc will be updated for sure.